dse_simulation/src/information_filter

```diff
@@ -0,0 +1,313 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import roslib
+import sys
+import rospy
+import numpy as np
+import datetime
+import time
+from dse_msgs.msg import PoseMarkers
+from std_msgs.msg import Float64MultiArray
+from std_msgs.msg import MultiArrayLayout
+from std_msgs.msg import MultiArrayDimension
+from dse_msgs.msg import InfFilterPartials
+from dse_msgs.msg import InfFilterResults
+from scipy.spatial.transform import Rotation as R
+
+from dse_lib import *
+
+roslib.load_manifest('dse_simulation')
+
+class information_filter:
+
+    # Set up initial variables
+    # Pass in the ID of this agent
+    def __init__(self, my_id):
+
+        # Define publishers and subscribers
+        self.pose_sub = rospy.Subscriber("/dse/pose_markers", PoseMarkers, self.measurement_callback)
+        self.results_sub = rospy.Subscriber("/dse/inf/results", InfFilterResults, self.results_callback)
+        self.inf_pub = rospy.Publisher("/dse/inf/partial", InfFilterPartials, queue_size=10)
+
+        # Define static variables
+        self.my_id = my_id
+        self.dim_obs = 6
+        self.dim_state = 12
+        self.dt = 0.1
+        self.t_last = rospy.get_time()
+        self.euler_order = 'zyx'
+
+        # Create information variables
+        self.inf_P = []
+        self.inf_x = []
+        #self.inf_Y = []
+        #self.inf_y = []
+        self.inf_I = []
+        self.inf_i = []
+        #self.inf_id_list = []
+        self.inf_id_obs = []
+        self.inf_id_comm = []
+        self.inf_id_list = [self.my_id]
+        self.inf_Y = 10000 * np.eye(self.dim_state, dtype=np.float64)
+        self.inf_y = 0.01 * np.transpose(1 * np.arange(1, self.dim_state+1, dtype=np.float64))[:, None]
+
+    # When the direct estimator or consensus returns the combined information variables
+    def results_callback(self, data):
+        self.inf_id_list = np.array(data.ids)
+        self.inf_Y = multi_array_2d_output(data.inf_matrix)
+        self.inf_y = multi_array_2d_output(data.inf_vector)
+
+    # When the camera sends a measurement
+    def measurement_callback(self, data):
+
+        # Compute the actual dt
+        self.dt = rospy.get_time() - self.t_last
+        self.t_last = rospy.get_time()
+
+        # Grab the tag poses from the camera
+        observed_poses = data.pose_array.poses
+        observed_ids = data.ids
+        n = 1 + len(observed_ids)
+
+        # update local values from the last time step
+        Y_11 = self.inf_Y
+        y_11 = self.inf_y
+        x_11 = np.linalg.inv(Y_11).dot(y_11)
+        P_11 = np.linalg.inv(Y_11)
+        id_list = self.inf_id_list
+
+        # If we find an ID that isn't currently known, add it
+        for id in observed_ids:
+            if not np.isin(id, id_list):
+                id_list = np.concatenate((id_list, [id]))
+                dim = len(id_list) * self.dim_state
+
+                Y_11_tmp = 10000 * np.eye(dim)
```

Comment:
What are we doing here? Why do we need temp? Why 10000? Please at least add a comment.